docs/source/installation/uv.md

```diff
@@ -0,0 +1,285 @@
+```bash
+sudo dnf install python3-devel pkg-config cairo-devel pango-devel
+```
+:::
+
+:::{tab-item} Arch Linux / pacman
+```bash
+sudo pacman -Syu base-devel cairo pango
+```
+:::
+
+::::
+
+As soon as the required dependencies are installed, you can create
+a Python project (feel free to change the name *manimations* used below
+to some other name) with a local environment containing Manim by running
+```bash
+uv init manimations
+cd manimations
+uv add manim
+```
+
+:::::
+
+::::::
+
+To verify that your local Python project is setup correctly
+and that Manim is available, simply run
+```bash
+uv run manim checkhealth
+```
+
+At this point, you can also open your project folder with the
+IDE of your choice. All modern Python IDEs (for example VS Code
+with the Python extension, or PyCharm) should automatically detect
+the local environment created by `uv` such that if you put
+```py
+import manim
+```
+into a new file `my-first-animation.py`, the import is resolved
+correctly and autocompletion is available.
+
+*Happy Manimating!*
+
+
+:::{dropdown} Alternative: Installing Manim as a global `uv`-managed tool
+If you have Manim projects in many different directories and you do not
+want to setup a local project environment for each of them, you could
+also install Manim as a `uv`-managed tool.
+
+See [`uv`'s documentation for more information](https://docs.astral.sh/uv/concepts/tools/)
+on their tool mechanism.
+
+To install Manim as a global `uv` tool, simply run
+```bash
+uv tool install manim
+```
+after which the `manim` executable will be available on your
+global system path, without the need to activate any virtual
+environment or prefixing your commands with `uv run`.
+
+Note that when using this approach, setting up your code editor
+to properly resolve `import manim` requires additional work, as
+the global tool environment is not automatically detected: the
+base path of all tool environments can be determined by running
+```
+uv tool dir
+```
+which should now contain a directory `manim` in which the appropriate
+virtual environment is located. Set the Python interpreter of your IDE
+to this environment to make imports properly resolve themselves.
+:::
+
+:::{dropdown} Installing Manim for a different version of Python
+In case you would like to use a different version of Python
+(for example, due to compatibility issues with other packages),
+then `uv` allows you to do so in a fairly straightforward way.
+
+When initializing the local Python project, simply pass the Python
+version you want to use as an argument to the `init` command:
+```
+uv init --python 3.12 manimations
+cd manimations
+uv add manim
+```
+To change the version for an existing package, you will need to
+edit two files contained in the root folder of your project:
+in `.python-version`, change the version number to the one you want
+to use (e.g., `3.12`). And secondly, in `pyproject.toml`, the
+`requires-python` entry needs to be updated such that your chosen
```

Comment:
I'm not sure about the `requires-python` entry, but `uv python pin` should take care of most of the stuff.

Reply:
Did some testing, `uv python pin` errors if it doesn't satisfy the `requires-python`.